rice-framework/krad-web-framework/src/main/java/org/kuali/rice/krad/web/filter/UserLoginFilter.java

```diff
@@ -166,28 +173,72 @@ private String getKualiSessionId(final Cookie[] cookies) {
     /**
      * establishes the backdoor user on the established user id if backdoor capabilities are valid.
      */
-    private void establishBackdoorUser(HttpServletRequest request) {
+    protected void establishBackdoorUser(HttpServletRequest request) {
         final String backdoor = request.getParameter(KRADConstants.BACKDOOR_PARAMETER);
-        if (StringUtils.isNotBlank(backdoor)) {
-            if (!getKualiConfigurationService().getPropertyValueAsString(KRADConstants.PROD_ENVIRONMENT_CODE_KEY)
-                    .equalsIgnoreCase(getKualiConfigurationService().getPropertyValueAsString(
-                            KRADConstants.ENVIRONMENT_KEY))) {
-                if (getParameterService().getParameterValueAsBoolean(KRADConstants.KUALI_RICE_WORKFLOW_NAMESPACE,
-                        KRADConstants.DetailTypes.BACKDOOR_DETAIL_TYPE, KewApiConstants.SHOW_BACK_DOOR_LOGIN_IND)) {
-                    try {
-                        KRADUtils.getUserSessionFromRequest(request).setBackdoorUser(backdoor);
-                    } catch (RiceRuntimeException re) {
-                        //Ignore so BackdoorAction can redirect to invalid_backdoor_portal
-                    }
-                }
-            }
+        UserSession userSession = KRADUtils.getUserSessionFromRequest(request);
+
+        if (StringUtils.isNotBlank(backdoor) && !isProductionEnvironment() && showBackdoorLogin() &&
+                isBackdoorAuthorized(userSession)) {
+
+            KRADUtils.getUserSessionFromRequest(request).setBackdoorUser(backdoor);
         }
     }
 
+    /**
+     * Determines if the Backdoor login form should be displayed
+     * @return True if the backdoor form should be displayed, False otherwise
+     */
+    private boolean showBackdoorLogin() {
+        return getParameterService().getParameterValueAsBoolean(KRADConstants.KUALI_RICE_WORKFLOW_NAMESPACE,
+                KRADConstants.DetailTypes.BACKDOOR_DETAIL_TYPE, KewApiConstants.SHOW_BACK_DOOR_LOGIN_IND);
+    }
+
+    /**
+     * Determines if the current server instance is a production environment
+     * @return True if the current server instance is a production environment, False otherwise
+     */
+    private boolean isProductionEnvironment() {
+        return getKualiConfigurationService().getPropertyValueAsString(KRADConstants.PROD_ENVIRONMENT_CODE_KEY)
+                .equalsIgnoreCase(getKualiConfigurationService().getPropertyValueAsString(KRADConstants.ENVIRONMENT_KEY));
+    }
+
     private void addToMDC(HttpServletRequest request) {
         MDC.put(MDC_USER, KRADUtils.getUserSessionFromRequest(request).getPrincipalName());
     }
 
+    /**
+     * Determines if the actual logged in user has backdoor permissions.
+     * @param uSession the current UserSession
+     * @return True if the user has permissions to user the backdoor function, False otherwise
+     */
+    private boolean isBackdoorAuthorized(UserSession uSession) {

```

Comment:
May want to make this and the other helpers protected in case we need to reference it from the IU extension of this class